pandas/core/internals.py

```diff
@@ -2592,6 +2577,44 @@ def set(self, locs, values, check=False):
 
         self.values[locs] = values
 
+    def _interpolate(self, method=None, index=None, values=None,
+                     fill_value=None, axis=0, limit=None,
+                     limit_direction='forward', inplace=False, downcast=None,
+                     mgr=None, **kwargs):
+        """ interpolate using scipy wrappers, adapted to datetime64 values"""
+
+        inplace = validate_bool_kwarg(inplace, 'inplace')
+        data = self.values if inplace else self.values.copy()
+
+        # only deal with floats
+        mask = isna(self.values)
+        if self.is_datetimetz:
+            # Convert to UTC for interpolation
+            data = data.tz_convert('UTC').values
+        if self.ndim > 1:
+            # DataFrame
+            data = np.atleast_2d(data)
+            mask = np.atleast_2d(mask)
+        data = data.astype(np.float64)
+        data[mask] = np.nan
+
```

Comment:
ths is a hodgepode. use the hierarchy to make this code readable. IOW put the datetime stuff in the dateimeblock.

Reply:
This is in the DatetimeBlock.  I'm confused.

Reply:
then why would you need is_datetime.

Reply:
this needs to be be way less if/then stuff.

Reply:
> then why would you need is_datetime.

It's specifically `is_datetimetz`; the tzaware case gets slightly different handling.  If you're suggesting a separate implementation for DatetimeTZBlock vs DatetimeBlock I guess that's OK, but pretty sure there'll be complaints about code duplication that way.